src/decorators.ts

```diff
@@ -48,31 +48,35 @@ export interface PropertyOptions {
 
 function createProperty(
     proto: any, name: string, options?: PropertyOptions): void {
-  const notify = options && options.notify || false;
-  const reflectToAttribute = options && options.reflectToAttribute || false;
-  const readOnly = options && options.readOnly || false;
-  const computed = options && options.computed || '';
-  const observer = options && options.observer || '';
-
-  let type;
-  if (options && options.hasOwnProperty('type')) {
-    type = options.type;
-  } else if (
-      (window as any).Reflect && Reflect.hasMetadata && Reflect.getMetadata &&
-      Reflect.hasMetadata('design:type', proto, name)) {
-    type = Reflect.getMetadata('design:type', proto, name);
-  } else {
-    console.error(
-        `A type could not be found for ${name}. ` +
-        'Set a type or configure Metadata Reflection API support.');
-  }
-
   if (!proto.constructor.hasOwnProperty('properties')) {
     Object.defineProperty(proto.constructor, 'properties', {value: {}});
   }
 
-  const finalOpts: PropertyOptions =
-      {type, notify, reflectToAttribute, readOnly, computed, observer};
+  const finalOpts: PropertyOptions = proto.constructor.properties[name] || {};
+
+  if (options) {
+    finalOpts.notify = options.notify || finalOpts.notify || false;
```

Comment:
The `||` here means that a `true` value would always win, regardless of the decorator order. I think the latter applied decorator should always win. Check `undefined` instead?

Reply:
yup i think i may just rewrite this getting-rather-lengthy block of `||` to explicit conditions